### ISSUE TITLE
Logging of IEnumerable<>/GetEnumerator Method => does not reset IEnumerator pointer

### ISSUE THREAD

**Issue by HeikoWitthoeft** [csnemes/tracer#22]
Hey,

I just found a bug which just driven me crazy, if you overload any IEnumerable (here a Dictionary to get it thread safe) and you enable tracer for the method (GetEnumerator) which really return the Enumerator the logging is incrementing the pointer but is not resetting it - so if you just iterate over the list with linq or any foreach the contained object is never returned. 

I think this can be easily adapted by extending the  GetRenderedFormat of the Adapter - I will just try it.

BR
Heiko

Update:
This does the magic: LoggerAdapter.cs - Line 439 add this:
```    
            else if (message is IEnumerator)
            {
                var retVal = _logger.Repository.RendererMap.FindAndRender(message);
                var enumerable = message as IEnumerator;
                enumerable.Reset();
                return retVal;
            }
```
I dont have a GIT setup here (we are still using SVN) - could you apply this or is there any downside? I just tested it here and it seems to work.

**Comment by HeikoWitthoeft** [comment]
I think i manged to fork / commit and add a pull request directly here without any tools ;) If i have done it wrong just tell me - im not that familiar with the environment here.

**Comment by csnemes** [comment]
Thanks for the fix. I released it in 1.3

**Comment by HeikoWitthoeft** [comment]
This does not seems to be the end of problems related to the tracing. I am now facing problems related to the IDataRecord interface. So Tracer gets the value and it is gone for the execution of the method itself. Right now I have no solution but i will update here as soon as...
